message_definitions/v1.0/ras_a.xml

```diff
@@ -10,7 +10,140 @@
     <!-- ALL the entries in the MAV_CMD enum have a maximum of 7 parameters -->
     <enum name="MAV_CMD">
       <description>Commands to be executed by the MAV. They can be executed on user request, or as part of a mission script. If the action is used in a mission, the parameter mapping to the waypoint/mission message is as follows: Param 1, Param 2, Param 3, Param 4, X: Param 5, Y:Param 6, Z:Param 7. This command list is similar what ARINC 424 is for commercial aircraft: A data format how to interpret waypoint/mission data. NaN and INT32_MAX may be used in float/integer params (respectively) to indicate optional/default values (e.g. to use the component's current yaw or latitude rather than a specific value). See https://mavlink.io/en/guide/xml_schema.html#MAV_CMD for information about the structure of the MAV_CMD entries</description>
-      <entry value="199" name="MAV_CMD_SET_POI_FOCUS" hasLocation="false" isDestination="false">
+      <entry value="40500" name="MAV_CMD_GO_THROUGH_PORTAL" hasLocation="true" isDestination="false">
+        <wip/>
+        <!-- This command is work-in-progress and it can therefore change. It should NOT be used in stable production environments. -->
+        <description>
+          Go through a portal. In an indoor exploration context, a portal represents an structural identifiable entry or pass point to start, stop or continue an exploration.
+          COMMAND_INT should be used so to set the MAV_FRAME and consequently, frame and coordinates of the portal position (MAV_FRAME_GLOBAL means global coordinates are being used,
+          while MAV_FRAME_LOCAL_NED means local NED coordinates are being used).
+        </description>
+        <param index="1" label="Portal ID" minValue="0" maxValue="4294967295" increment="1">ID of the portal. If the ID is unknown or not deterministic, set it to UINT32_MAX and use params 5, 6 and 7 to define the local or global position of the portal.</param>
```

Comment:
What does "not deterministic" mean?

Reply:
Also, since this is always a float, UINT32_MAX doesn't make sense. Also, shouldn't the maxValue be the largest integer-safe 32-bit float? Which is 16,777,216 I believe.

Reply:
@dskyle-shieldai Good points. If you use the restricted values, can you represent the POIs you might get?

Reply:
I don't think it matters. For example [here](https://github.com/Dronecode/air-iop-definitions/pull/4/files#diff-48eda1727976f638e3e4752aab8c9ce4818837f66699409fa7ef0f4db4050b79R30-R32), apparently it is possible to concatenate a big-endian 64-bits UID with these two first fields using a 32-bit integer of these fields. I believe it doesn't matter this is defined as a float as long as you cast the fields properly on each side. The only thing it matters is that this is a 32-bit fields. It doesn't necessarily matter the type. Check also [this](https://github.com/mavlink/mavlink/issues/1523).

Reply:
What is "casting the fields properly"? Casting an INT32_MAX to a float in the normal way results in a value equivalent to unsigned 0x80000000, which when cast back to an int, is implementation defined, but typically -1.

Are you suggesting that we memcpy integers into the space for the float, then memcpy them back into an int? I'm not sure that's reliable, either: https://stackoverflow.com/questions/57355462/is-it-guaranteed-that-the-copy-of-a-float-variable-will-be-bitwise-equivalent-to

I believe the PR you linked is similarly flawed, and should not take that approach.

Reply:
> Are you suggesting that we memcpy integers into the space for the float, then memcpy them back into an int? I'm not sure that's reliable, either: https://stackoverflow.com/questions/57355462/is-it-guaranteed-that-the-copy-of-a-float-variable-will-be-bitwise-equivalent-to

@dskyle-shieldai Yes he is. This approach has been used for 15 years for parameters so perhaps there are cases where it is unreliable in theory, but we "know" it hasn't been a problem in MAVLink.
A description of what happens in this case is in https://mavlink.io/en/services/parameter.html#parameter-encoding

The definition of the [command protocol](https://mavlink.io/en/services/command.html) in mavlink does not address this case - it only states that params 5, 6 may not contain things that can only be sent in floats. 

@TSC21 Obviously you "can" do this. You'd need to document the conversion.
Since you have to split the POI anyway have you considered perhaps it would be easier to split it into smaller chunks that can just be copied into the float directly. You'll lose some spare slots of course, but if you're fairly sure this is stable ...

Reply:
Indeed, if that's the intent, it needs to be much more clearly documented. But I don't think it _should_ be the intent. 2^24 portal IDs is plenty for now, and we can contemplate a better solution for integer arguments in general for RAS-A 2.0.

I still believe my suggestion for this spec remains the best available: cap the max at 2^24, which is integer-safe for essentially all implementations of float, and use NaN for the invalid/default.

Reply:
I have no strong opinion; either way works, just needs to be unambiguous.